### PR TITLE
feat/stagingBuildStatus/storybook

### PR DESCRIPTION
**Pull request by kishore03109**
## Problem

Storybooks for the 3 states regarding the site status.

### CODE REVIEW

**Comment by kishore03109** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #1635** <a href="https://app.graphite.dev/github/pr/isomerpages/isomercms-frontend/1635" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/isomerpages/isomercms-frontend/1635?utm_source=stack-comment).

File: src/types/stagingBuildStatus.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export const statusStates = {
```
Comment (dcshzj) [inline src/types/stagingBuildStatus.ts:1]: nit: should be caps

File: src/types/stagingBuildStatus.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  pending: "PENDING",
+  ready: "READY",
+  error: "ERROR",
+} as const
+export type buildStatus = typeof statusStates[keyof typeof statusStates]
+
+export interface StagingBuildStatus {
+  commit: string
+  status: buildStatus
+  timeLastSaved: number
```
Comment (dcshzj) [inline src/types/stagingBuildStatus.ts:12]: nit: `commit` and `timeLastSaved` doesn't seem to be returned from the backend (was referencing https://github.com/isomerpages/isomercms-backend/pull/1008)

File: src/components/Header/StatusBadge.tsx [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    timeText = `${timeLastSavedInMin} minute${
+      timeLastSavedInMin === 1 ? "" : "s"
+    } ago`
+  } else if (timeLastSavedInMin < 60 * 60) {
+    const hours = Math.floor(timeLastSavedInMin / 60)
+    timeText = `${hours} hour${hours === 1 ? "" : "s"} ago`
+  } else {
+    const days = Math.floor(timeLastSavedInMin / 60 / 24)
+    timeText = `${days} day${days === 1 ? "" : "s"} ago`
+  }
```
Comment (dcshzj) [inline src/components/Header/StatusBadge.tsx:53]: Might be good to put inside dateUtils or potentially enhance `convertUtcToTimeDiff`.

File: src/components/Header/StatusBadge.tsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+  switch (status) {
+    case "READY":
+      headingText = `All saved edits from ${timeText} are on staging`
+      bodyText = "Click on 'Open staging' to take a look."
+
+      Icon = (props: IconBaseProps) => (
+        <BiCheckCircle {...props} color="#0F796F" />
```
Comment (dcshzj) [inline src/components/Header/StatusBadge.tsx:60]: nit: can use `utility.feedback.success`?

File: src/components/Header/StatusBadge.tsx [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+      break
+    case "READY":
+      displayText = "Staging site is ready"
+      colourScheme = "success"
+      dotColor = "#0F796F"
+      break
+    case "ERROR":
+      displayText = "Staging site is out of date"
+      colourScheme = "warning"
+      dotColor = "#FFDA68"
```
Comment (dcshzj) [inline src/components/Header/StatusBadge.tsx:117]: nit: can upgrade the dotColors to tokens if possible

File: src/components/Header/StatusBadge.tsx [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+  return (
+    <Box position="relative">
+      <Popover trigger="hover">
+        <PopoverTrigger>
+          <Badge colorScheme={colourScheme} variant="subtle" cursor="default">
+            <GoDotFill size="1rem" color={dotColor} />
+            <Text ml="0.5rem" mr="0.5rem">
+              {displayText}
+            </Text>
+            <BsFillQuestionCircleFill color="#454953" />
```
Comment (dcshzj) [inline src/components/Header/StatusBadge.tsx:132]: nit: can use `grey.700`?

**Review by dcshzj — APPROVED** [review]

File: src/types/stagingBuildStatus.ts (hunk 2 above)
Comment (dcshzj) [reply src/types/stagingBuildStatus.ts:12]: Lol ignore, saw that it was fixed in #1636.

Comment (kishore03109) [reply src/types/stagingBuildStatus.ts:12]: ya ps shifted my approach halfway

File: src/components/Header/StatusBadge.tsx (hunk 3 above)
Comment (kishore03109) [reply src/components/Header/StatusBadge.tsx:53]: oh i removed it in the end, but will add this back in dateUtils

File: src/components/Header/StatusBadge.tsx (hunk 4 above)
Comment (kishore03109) [reply src/components/Header/StatusBadge.tsx:60]: wait ah could you use this for icons? it didnt work for me idk why icons screws up

File: src/components/Header/StatusBadge.tsx (hunk 5 above)
Comment (kishore03109) [reply src/components/Header/StatusBadge.tsx:117]: same here

File: src/components/Header/StatusBadge.tsx (hunk 4 above)
Comment (dcshzj) [reply src/components/Header/StatusBadge.tsx:60]: I think can use Chakra's `<Icon>` then the `as` prop with the color tokens

Comment (kishore03109) [reply src/components/Header/StatusBadge.tsx:60]: orh right ok uhm next monday release bah

**Comment by kishore03109** [comment]
@dcshzj colour stuff will settle on monday

**Comment by dcshzj** [comment]
@kishore03109 yeah can not blocker

**Comment by kishore03109** [comment]
### Merge activity

* **Nov 2, 3:18 AM**: @kishore03109 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/isomerpages/isomercms-frontend/1635).

**Pull request merged**